manifests/package.pp

```diff
@@ -153,9 +153,20 @@
   }
 
   if ($elasticsearch::package_provider == 'package') {
-    package { 'elasticsearch':
-      ensure => $package_ensure,
-      name   => $elasticsearch::_package_name,
+    # You cannot use "mark" property while "ensure" is one of ["absent", "purged", "held"]
+    if $package_ensure in ['absent', 'purged', 'held'] {
+      package { 'elasticsearch':
+        ensure => $package_ensure,
+        name   => $elasticsearch::_package_name,
+      }
+    } else {
+      # https://puppet.com/docs/puppet/7/types/package.html#package-attribute-mark
+      # Set to hold to tell Debian apt/Solaris pkg to hold the package version
+      package { 'elasticsearch':
+        ensure => $package_ensure,
+        name   => $elasticsearch::_package_name,
+        mark   => ($elasticsearch::autoupgrade ? { true  => 'none', false => 'hold', }),
```

Comment:
Not sure if re-using `$elasticsearch::autoupgrade` is a good idea. Of course if one wants updates to be automatically installed holding the package does not make sense, but otherwise one will probably want to know if updates are available.

What do you think about adding a new parameter like `$elasticsearch::package_hold`?

Reply:
Ok but i will do this modification when i come back after my holidays ;) (i have only my ipad actually)

Reply:
Done